platform/fabricx/core/finality/listenermanager.go

```diff
@@ -18,29 +19,63 @@ import (
 	"github.com/hyperledger/fabric-x-committer/api/protonotify"
 )
 
+// ListenerManager defines the interface for managing finality listeners for transactions.
 type ListenerManager interface {
 	AddFinalityListener(txID driver.TxID, listener fabric.FinalityListener) error
 	RemoveFinalityListener(txID driver.TxID, listener fabric.FinalityListener) error
-	Listen(ctx context.Context) error
 }
 
+// ListenerManagerProvider defines the interface for creating new ListenerManager instances
+// for specific network and channel combinations.
 type ListenerManagerProvider interface {
-	NewManager(ctx context.Context, network, channel string) (ListenerManager, error)
+	NewManager(network, channel string) (ListenerManager, error)
 }
 
-func NewListenerManagerProvider(fnsp *fabric.NetworkServiceProvider, configProvider config.Provider) ListenerManagerProvider {
-	return &listenerManagerProvider{
+// NewListenerManagerProvider creates a new instance of the Provider, which implements ListenerManagerProvider.
+func NewListenerManagerProvider(fnsp *fabric.NetworkServiceProvider, configProvider config.Provider) *Provider {
+	return &Provider{
 		fnsp:           fnsp,
 		configProvider: configProvider,
+		managers:       make(map[string]ListenerManager),
 	}
 }
 
-type listenerManagerProvider struct {
+type Provider struct {
```

Comment:
I believe it would be nice to describe the provider and how it is supposed to be used (call Initialize before NewManager) ...